src/commands/mod.rs

```diff
@@ -212,14 +212,29 @@ impl CommandParser {
         }
     }
 
-    fn _next_integer(&mut self) -> Result<i64, CommandParserError> {
+    fn next_integer(&mut self) -> Result<i64, CommandParserError> {
         let frame = self
             .parts
             .next()
             .ok_or_else(|| CommandParserError::EndOfStream)?;
 
         match frame {
             Frame::Integer(i) => Ok(i),
+            Frame::Simple(string) => {
+                string
+                    .parse::<i64>()
+                    .map_err(|_| CommandParserError::InvalidFrame {
+                        expected: "parseable i64 frame".to_string(),
+                        actual: Frame::Simple(string),
+                    })
+            }
+            Frame::Bulk(bytes) => str::from_utf8(&bytes[..])
+                .map_err(CommandParserError::InvalidUTF8String)?
+                .parse::<i64>()
+                .map_err(|_| CommandParserError::InvalidFrame {
+                    expected: "parseable i64 frame".to_string(),
+                    actual: Frame::Bulk(bytes),
```

Comment:
We need to revisit the error messages at some point.

Reply:
We have to revisit all error messages 😅